api/controllers/auth.js

```diff
@@ -44,6 +50,48 @@ const registration = async (req, res, next) => {
   }
 };
 
+const reset = async (req, res, next) => {
+  const { email } = req.body;
+  try {
+    const v = await emailSchema.validate({ email });
+    const a = v && (await services.resetPasswordRequest({ email }));
+    let status = a.status ? a.status : 200;
+    let response = a.messageId ? 'Mail was sent' : a.message;
+    res.status(status).send(response);
+  } catch (e) {
+    next(e);
+  }
+  //
+};
+
+const resetApprovementPassword = async (req, res, next) => {
+  const { linkId } = req.params;
+  try {
+    const UserId = await services.resetPasswordApprove({
+      linkId: decodeURIComponent(linkId),
+    });
+    if (UserId) {
+      res.status(200).send({
+        status: 200,
+        message: { content: 'Enter New Password', UserId },
+      });
+    }
+  } catch (e) {
+    next(e);
+  }
```

Comment:
Why you always use next to throwing errors? We have special middleware for this, so you can do something like this: `throw new Error('My error")` and after we do custom error classes we will change it

Reply:
Besause we have error middleware

Reply:
So? Without `next()` it will works.